include/hip/hcc_detail/hip_runtime.h

```diff
@@ -235,8 +235,43 @@ __device__ int __hip_move_dpp(int src, int dpp_ctrl, int row_mask, int bank_mask
 
 #endif  //__HIP_ARCH_GFX803__ == 1
 
-__host__ __device__ int min(int arg1, int arg2);
-__host__ __device__ int max(int arg1, int arg2);
+__host__ __device__ int          min(         int arg1,          int arg2);
+__host__ __device__ unsigned int min(         int arg1, unsigned int arg2);
```

Comment:
This signature looks incorrect to me. ```min(int,unsigned int)``` should be ```int``` in my opinion. For example ```min(-100,20)``` is -100 which is ```int```. Same applies to all other min variants that compare a signed and unsigned value.

Reply:
yeah, in theory we should not need to have min/max with one arg signed and the other unsigned :)

The reason I needed to add the sign mismatch overloads is because this scenario manifests itself in the TF codebase.  I saw the same overloads on the CUDA side, and am assuming they added those overloads atleast in part to address the scenario in the TF codebase. It would have been nicer if the TF codebase was fixed instead, but we have no control over that.

The return type was chosen to match what CUDA has. Either choice is problematic as it will not cover the entire range of one of the two args. We can switch the return type to signed if matching equivalent CUDA funcs is not that important.

Reply:
For the sign mis-match overloads, the CUDA implementation simply casts the signed argument to unsigned, and then calls the unsigned overload, which is why the return type is also unsigned

Reply:
I had rather change the return type to signed as well for those sign mismatch overloads, so if CUDA compatibility is not a requirement / preference, then let me know and I will update the PR accordingly

Reply:
actually come to think of it, returning `signed` for `min` and `unsigned` for max` is the correct choice. That is what I will go with, if the HIP signature can be different than CUDA